### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by dmytrostriletskyi**
Documentation located in `openedx/core/djangoapps/edx_global_analytics/README.md`:

1. Common things as description.
2. Explanation statistics fetching and sending flow.
3. How statistics bunch depends on statistics level.
4. Application settings.
5. How to run tests.
6. Development details (celery settings, celery command).

TODO:

- [x] Production things.

### CODE REVIEW

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+`EdX global analytics` is part of edx-platform as Django application. It has arisen out of [Open edX](https://open.edx.org)
+requirement to collect edx-platform installations statistics data from all over the world.
+
+So one at 24 hours this application fetches and sends edX installation statistics to [Acceptor](https://github.com/raccoongang/acceptor) developed by @raccoongang team.
+Acceptor is required to be able to collect, visualize and process this data. In general, it provides possibilities
+for analysing trends of platform usage and users engagement in e-learning process per country and globally around the world.
+It is usually carried out via graphs, world map and activity metrics.
+
+## The way sent data looks like on Acceptor
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: We remove Acceptor and rename to Olga

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+![acceptor_world_map](https://user-images.githubusercontent.com/22666467/27970248-d015fd9c-6356-11e7-906c-8c052cd65b08.png)
+
+## Types of statistics sent, and dependence on statistics level
+
+Firstly, `statistic level` is a string constant in settings (`lms.env.json`) which regulates size of bunch platform sends.
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: Please add statement, that this is optional and all will work on paranoid level w/o any changes to settings

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
+# edX global analytics
+
+`EdX global analytics` is part of edx-platform as Django application. It has arisen out of [Open edX](https://open.edx.org)
+requirement to collect edx-platform installations statistics data from all over the world.
+
+So one at 24 hours this application fetches and sends edX installation statistics to [Acceptor](https://github.com/raccoongang/acceptor) developed by @raccoongang team.
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: one -> once

Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: you need to add yourself to authors file of edx-platform as well (See edx docs).

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+Acceptor is required to be able to collect, visualize and process this data. In general, it provides possibilities
+for analysing trends of platform usage and users engagement in e-learning process per country and globally around the world.
+It is usually carried out via graphs, world map and activity metrics.
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: usually -> currently

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 1 above)
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: Wrong wording, please rephrase.

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    'collect_stats': {
+        'task': 'openedx.core.djangoapps.edx_global_analytics.tasks.collect_stats',
+        'schedule': crontab(hour=0, minute=random.randint(1, 59)),
+    }
+}
+```
+
+Run celery task with command:
+
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: Add block "architecture" where shortly explain architecture of the project, for example, that it is use celery, now authorizes and how many requests per days are sent.

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+1. Access token for Acceptor server.
+2. Active students amount per last calendar day, week and month*.
+3. Courses amount.
+
+Second `Enthusiast` level and the last one with `enthusiast` constant extends `Paranoid` level with:
+1. Platform URL and platform name.
+2. Platform's latitude and longitude.
+3. Active students per country accordance for last calendar day.
+
+\*active student is a student whose last login datetime value is included in calendar duration.
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: "calendar duration" ?

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+Platform's administrator is able to configure `EdX global analytics` settings in `lms.env.json`.
+This settings already exist after default platform installation, admin just needs to change it.
+
+Settings context:
+```
+...,
+"OPENEDX_LEARNERS_GLOBAL_ANALYTICS": {
+    "CELERY_TIMEZONE": "Europe/Kiev",
+    "OLGA_ACCEPTOR_PERIODIC_TASK_URL": "",
+    "OLGA_ACCEPTOR_PERIODIC_TASK_URL_LOCAL_DEV": "http://192.168.1.10:7000",
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: this is not needed for admin, by default, move it to the section local development please

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    "PLATFORM_CITY_NAME": "Kiev",
+    "STATISTICS_LEVEL": "enthusiast"
+}, ...
+```
+
+`CELERY_TIMEZONE`: Time zone in edx-platform is enabled by default, platform's administrator can set time zone in `lms.env.json` for `TIME_ZONE` key. `EdX global analytics` supports own time zone setting — time zone platform will fetch and send statistics to Acceptor.
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: Why we need our own timezone? Maybe we can remove this at all, please explain, i do not understand it.

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+CELERYBEAT_SCHEDULE = {
+    """
+    'collect_stats' is the celery periodic task that gathers information about the
+    students amount, geographical coordinates of the platform, courses amount and
+    makes a POST request with the data to the appropriate service.
+    """
+
+    'collect_stats': {
+        'task': 'openedx.core.djangoapps.edx_global_analytics.tasks.collect_stats',
+        'schedule': crontab(hour=0, minute=random.randint(1, 59)),
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: and for production?

File: openedx/core/djangoapps/edx_global_analytics/migrations/0001_initial.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+    ]
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/migrations/0001_initial.py]: Why this is in documentation PR?

File: openedx/core/djangoapps/edx_global_analytics/tests/test_acceptor_api_usage.py [hunk 11]
```diff
@@ -0,0 +1,185 @@
+"""
+Tests acceptor API usage by edx_global_analytics application functions aka utils.
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/tests/test_acceptor_api_usage.py]: Why this is in documentation PR?

File: openedx/core/djangoapps/edx_global_analytics/migrations/0001_initial.py (hunk 10 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/migrations/0001_initial.py]: I did bad checkout. Sorry.

File: openedx/core/djangoapps/edx_global_analytics/tests/test_acceptor_api_usage.py (hunk 11 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/tests/test_acceptor_api_usage.py]: The same as above.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 3 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > one -> once

Done.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 4 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > usually -> currently

Done.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 1 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > We remove Acceptor and rename to Olga



> Wrong wording, please rephrase.

Done.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 2 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > Please add statement, that this is optional and all will work on paranoid level w/o any changes to settings

Done.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 6 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > "calendar duration" ?

Fix to `periods`.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 8 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: Discussed.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 5 above)
Comment (DmytryiStriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > Add block "architecture" where shortly explain architecture of the project, for example, that it is use celery, now authorizes and how many requests per days are sent.

Done.

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
+# edX global analytics
+
+`EdX global analytics` is part of edx-platform as Django application. It has arisen out of [Open edX](https://open.edx.org)
+requirement to collect edx-platform installations statistics data from all over the world.
+
+So once at 24 hours this application fetches and sends edX installation statistics to [OLGA](https://github.com/raccoongang/OLGA) developed by @raccoongang team.
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: Remove "so" please.

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+OLGA is required to be able to collect, visualize and process this data. In general, it provides possibilities
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: Remove "In general" please

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+for analysing trends of platform usage and users engagement in e-learning process per country and globally around the world.
+It is currently carried out via graphs, world map and activity metrics.
+
+## Sent statistics to OLGA looks like
+
+![olga_world_map](https://user-images.githubusercontent.com/22666467/27970248-d015fd9c-6356-11e7-906c-8c052cd65b08.png)
+
+## Types of statistics sent, and dependence on statistics level
+
+Firstly, `statistic level` is a string constant in settings (`lms.env.json`) which regulates size of bunch platform sends.
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: rephrase "size of bunch" please

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+It is optional point, if you leave blank string there, it will work on `Paranoid` level (described below).
+
+First `Paranoid` level with paranoid constant allows platform to transfer:
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: remove "First" and "Second" please

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+1. Access token for OLGA server.
+2. Active students amount per last calendar day, week and month*.
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: we need * at the end?

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 7 above)
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: PLease add "settings for production" section

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+## Application's architecture
+
+### Celery
+
+Celery run one task once per day at time between `00:01` and `00:59` for sending statistics.
+
+Celery settings located in `lms/envs/aws.py`:
+
+```
+For scheduling tasks, entries can be added to this dict.
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: This is optional?

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 12 above)
Comment (dmytrostriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > Remove "so" please.

Done.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 13 above)
Comment (dmytrostriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > Remove "In general" please

Done.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 14 above)
Comment (dmytrostriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > rephrase "size of bunch" please

Done.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 15 above)
Comment (dmytrostriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > remove "First" and "Second" please

Done.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 16 above)
Comment (dmytrostriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > we need * at the end?

Extended 2nd point of `Paranoid` instead `*`.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 17 above)
Comment (dmytrostriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > This is optional?

Nope, it is require for task. It is `Celery` settings. Rephrased.

File: openedx/core/djangoapps/edx_global_analytics/README.md (hunk 7 above)
Comment (dmytrostriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > PLease add "settings for production" section

I will push content after release, when i will know all of that.

File: openedx/core/djangoapps/edx_global_analytics/README.md [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+
+Run celery task locally with command:
+
+```
+./manage.py lms celery worker -B --settings=devstack_with_worker
+```
+
+## Production
+
+[production settings and things]
```
Comment (auraz) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: Why empty here?

Comment (dmytrostriletskyi) [inline openedx/core/djangoapps/edx_global_analytics/README.md]: > Why empty here?

I cannot write production points before I push project to production environment.

**Pull request merged**